src/resources/views/crud/fields/google_map.blade.php

```diff
@@ -0,0 +1,218 @@
+    >
+
+    <div style="width: 100%;height: 400px" id="map_{{$field['name']}}"></div>
+
+    <span class="btn btn-dark btn-block" id="locationButton_{{$field['name']}}">
+        <i class="la la-map-pin"></i> {{ __('Get My Location') }}
+    </span>
+</div>
+{{-- HINT --}}
+@if (isset($field['hint']))
+    <p class="help-block">{!! $field['hint'] !!}</p>
+@endif
+@include('crud::fields.inc.wrapper_end')
+{{-- Note: you can use  to only load some CSS/JS once, even though there are multiple instances of it --}}
+
+{{-- ########################################## --}}
+{{-- Extra CSS and JS for this particular field --}}
+{{-- If a field type is shown multiple times on a form, the CSS and JS will only be loaded once --}}
+
+@if ($crud->fieldTypeNotLoaded($field))
+    @php
+        $crud->markFieldTypeAsLoaded($field);
+    @endphp
+
+    {{-- FIELD CSS - will be loaded in the after_styles section --}}
+    @push('crud_fields_styles')
+        <style>
+            .ap-input-icon.ap-icon-pin {
+                right: 5px !important;
+            }
+
+            .ap-input-icon.ap-icon-clear {
+                right: 10px !important;
+            }
+
+            .pac-container {
+                z-index: 1051;
+            }
+        </style>
+    @endpush
+
+    {{-- FIELD JS - will be loaded in the after_scripts section --}}
+    @push('crud_fields_scripts')
+        <script>
+            var list = document.getElementsByTagName('script');
+            var i = list.length, flag = false;
+            while (i--) {
+                if (list[i].src === 'https://maps.googleapis.com/maps/api/js?v=3&key={{ $field['api_key'] ?? config('services.google_places.key') }}&libraries=places&callback=initGoogleAddressAutocomplete') {
+                    flag = true;
+                    break;
+                }
+            }
+
+            // if we didn't already find it on the page, add it
+            if (!flag) {
+                var tag = document.createElement('script');
+                tag.src = 'https://maps.googleapis.com/maps/api/js?v=3&key={{ $field['api_key'] ?? config('services.google_places.key') }}&libraries=places&callback=initGoogleAddressAutocomplete';
+                document.getElementsByTagName('body')[0].appendChild(tag);
+            }
+        </script>
+        <script>
+
+
+            function bpFieldInitGoogleMapElement(element) {
+                try {
+                    var $addressConfig = element.data('google-address');
+
+                    var $field = $('[name="' + $addressConfig.field + '"]');
+                    if ($field.val().length) {
+                        var existingData = JSON.parse($field.val());
+                        var latlng = new google.maps.LatLng(existingData.lat, existingData.lng)
+                        var isDefault = false;
+                    } else {
+                        var latlng = new google.maps.LatLng(29.97917348330557, 31.134265757949812)
```

Comment:
This should definitely be made configurable.

```suggestion
                        var lat = @json(config('services.google-places.default-map-center.lat', 29.97917))
                        var lng = @json(config('services.google-places.default-map-center.lng', 31.13426))
                        var latlng = new google.maps.LatLng(lat, lng)
```